linebot/account_link_test.go

```diff
@@ -83,7 +83,15 @@ func TestIssueLinkToken(t *testing.T) {
 		w.Write(tc.Response)
 	}))
 	defer server.Close()
-	client, err := mockClient(server)
+
+	dataServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer r.Body.Close()
+		w.WriteHeader(404)
+		w.Write([]byte(`{"message":"Not found"}`))
```

Comment:
How about calling t.Error in this handler?
we want to know dataServer is not called

Reply:
3c28b82 applies changes to make the intention clearer.

Reply:
I’m not sure where this p-r is going.

Just to make sure, are you saying that this should return the HTTP status code of 200 and the expected response? And only leave error state for the wrong API endpoint call? The tests I added kind of focus on the behavior of target API. It is somewhat like “If a wrong endpoint is called, then an unexpected response is returned and hence following test items fail.”

Honestly, the more I read tests in this repository I get more confused. For example, subtests are executed in a loop, but a `httptest.Server` is constructed outside of the loop and it still refers to `testing.T` which does not belong to the subtest. So the nested output of the test would look pretty weird.
And some request methods have benchmark tests but some do not. The benchmark actually points to a mock server so the main differences would be the time elapsed for payload serialization and deserialization. Probably this is because all `XxxCall`s currently have `Do` method so it seemed like all request methods needed to have benchmark tests as long as one was created for a certain method, but I am not sure about the consequence.

I started with a minimal code change before the first code review because a) this had to be merged prior to the notified date and b) some pull requests with potential conflicts were under review. But now it involves changes for overall architecture including the signature changes of `mockClient` while the deadline is getting close.

Probably I should close this p-r and let someone familiar with this repository’s policy take over so this issue can be handled in a short time.